### PR TITLE
AK+LibWeb: First steps for PowerPC CPU architecture support

### PR DESCRIPTION
**Pull request by sideeffect42**
This PR adds `AK_IS_ARCH` defines for ppc/ppc64/ppc64le and user agent CPU strings for 32-bit and 64-bit PowerPC.

For the 64-bit PowerPC user agent I followed the `RISC-V 64` pretty-printed example instead of using `ppc64(le)` like `x86_64` or `AArch64`.

These changes are not complete, only the first step to enable PowerPC support in Ladybird.

LibJS also needs to be updated (sign-extension of pointers) for getting little-endian ppc64le to work. (I didn't do it in this PR because I don't know how to do it correctly.)

For big-endian support in general more work is needed (#395, ...).

### CODE REVIEW

File: AK/Platform.h [hunk 1]
```diff
@@ -39,6 +39,23 @@
...
+#    define AK_IS_ARCH_PPC64LE() 1
+#else
+#    define AK_IS_ARCH_PPC64LE() 0
+#endif
+
+#if defined(__ppc__) || defined(__PPC__) || defined(__powerpc__) || defined(__powerpc) || defined(__POWERPC__)
+#    define AK_IS_ARCH_PPC() 1
+#else
+#    define AK_IS_ARCH_PPC() 0
+#endif
```
Comment (ADKaster) [inline AK/Platform.h:57]: we don't teeechnically support any 32-bit architectures very well. wasm32 mostly works, i686 used to work, but we aren't testing those anymore in this repository's CI. so they've probably bit-rotted already.

If you don't have a strong reason to support ppc 32-bit, I'd rather not add support for it.

**Comment by Hendiadyoin1** [comment]
Also note that a lot of code will break on big endian systems, and the few bit fields we have in this code base will also need adjusting (even on ppcle because they decided to ise the BE layout even on LE), same may go for the JS value nan-packing

Tldr: Just adding nicer target defines is not enough

File: AK/Platform.h (hunk 1 above)
Comment (sideeffect42) [reply AK/Platform.h:57]: So far this is just filling in missing values to get rid of compiler errors, I didn't do any investigation on the feasibility of Ladybird on ppc32.
My primary interest is getting Ladybird to run on ppc64(be), first.

Since ppc32 is historically big-endian only (except for some 90's experiments with WinNT) the feasibility of ppc32 support will have to be seen after big-endian support is implemented.

As far as strong reasons go: portability.
There is quite an active community around ppc and what is an architecture worth these days without a web browser :-)

I understand that the project's goal is to get a modern web browser going and 32-bit support may appear like an unnecessary blocker.
I would still like add these definitions, because they don't hurt.

**Comment by sideeffect42** [comment]
@Hendiadyoin1 as I have written in the OP this is just a "first steps" pull.
ppc64le seems to work fairly well with these adjustments and an arbitrary choice for pointer extension.

ppc64 requires more work, as I had written. I did do some work to make the parts which produce compile-time errors endian-agnostic.
This will be my next PR.
But there is still more work to be done and if you could give me some pointers or assistance in poking the right things, this would be very appreciated.

**Comment by sideeffect42** [comment]
The test suite on ppc64le runs through with all passes except for some Intl errors in the JS test suite because of some differences between the system ICU 72 and the test cases which expect ICU 74 output (e.g. #411).

**Review by ADKaster — APPROVED** [review]
